pilot/pkg/features/experimental.go

```diff
@@ -165,15 +165,23 @@ var (
 	EnableGatewayAPIGatewayClassController = env.Register("PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER", true,
 		"If this is set to true, istiod will create and manage its default GatewayClasses").Get()
 
-	// EnableHBONE provides a global Pilot flag for enabling HBONE.
-	// Generally, this could be a per-proxy setting (and is, via ENABLE_HBONE node metadata).
-	// However, there are some code paths that impact all clients, hence the global flag.
-	// Warning: do not enable by default until endpoint_builder.go caching is fixed (and possibly other locations).
-	EnableHBONE = env.Register(
-		"PILOT_ENABLE_HBONE",
+	// EnableHBONEListening provides a global Pilot flag for enabling listening on HBONE.
+	EnableHBONEListening = env.Register(
```

Comment:
Should these be control plane flags, or supersede ISTIO_META_HBONE?  Is it possiblewe'd want different behavior for each of these for gateways/sidecars?

Reply:
I suppose they can be per-pod. TBH I don't see much reason you would want to do it per pod but it doesn't hurt.

The need for the global control plane flag is now obsolete

Reply:
I'm worried about too many options here. Is there any scenario where user would NOT want to listen HBONE ? 
If they enabled ztunnel or any other ambient option and they have ztunnel on client side - I can't imagine user opting to break themselves. 

The real option is if the client ztunnel will talk with the ztunnel on the server side (where the sidecar lives), i.e. sandwich on server side - or will do HBONE directly with Envoy sidecar. As in my previous comment, if user has
opted for ambient for this client-server communication - it seems even more clear that their intent is to use ztunnel. Since this is not yet ready - we can keep listening HBONE until Envoy can handle HAPROXY properly, 
 but I can't see any reason for Envoy handling mTLS and L4 when user has opted to ambient.